File: front-end/Mobile/src/models/authorization/RolePermission.js
import BaseModel from '../BaseModel';

export default class RolePermission extends BaseModel {
  constructor(data = {}) {
    super(data);
    this.roleId = data.role_id || null;
    this.permissionId = data.permission_id || null;
    this.assignmentDate = data.assignment_date || null;
  }

  static fromApi(data) {
    if (!data) return null;
    return new RolePermission(data);
  }

  toApi() {
    return {
      ...super.toApi(),
      role_id: this.roleId,
      permission_id: this.permissionId,
      assignment_date: this.assignmentDate,
    };
  }
}

File: back-end/02-ms-authorization/src/main/java/com/faceattend_edu/authorization_service/domain/model/RolePermission.java
package com.faceattend_edu.authorization_service.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class RolePermission {
    private Integer roleId;
    private Integer permissionId;
    private Instant assignmentDate;
    private Instant createdAt;
    private Instant updatedAt;
    private Instant deletedAt;
    private UUID createdBy;
    private UUID updatedBy;
    private UUID deletedBy;
    private long rowVersion;

    public void touchCreated() {
        if (assignmentDate == null) assignmentDate = Instant.now();
        if (createdAt == null) createdAt = Instant.now();
        if (rowVersion == 0) rowVersion = 1L;
    }
}
